java.hints/src/org/netbeans/modules/java/hints/errors/VarCompDeclaration.java

```diff
@@ -0,0 +1,177 @@
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) throws Exception {
+            TreePath statementPath = ctx.getPath();
+            Tree parent = statementPath.getParentPath().getLeaf();
+            List<? extends StatementTree> statements;
+            switch (parent.getKind()) {
+                case BLOCK:
+                    statements = ((BlockTree) parent).getStatements();
+                    break;
+                //Todo : Handle CASE scenario
+                /*case CASE:
+                    statements = ((CaseTree) parent).getStatements();
+                    break;*/
+                default:
+                    throw new IllegalStateException(parent.getKind().name());
+            }
+            WorkingCopy wc = ctx.getWorkingCopy();
+            TreeMaker make = wc.getTreeMaker();
+            int pos = statements.indexOf(statementPath.getLeaf());
+            List<StatementTree> newStatements = new ArrayList<>();
+            if (pos > 0) {
+                newStatements.addAll(statements.subList(0, pos));
+            }
+
+            int current = 0;
+            for (current = pos; current < statements.size(); current++) {
+                StatementTree t = (StatementTree) statements.get(current);
+                if (t instanceof VariableTree) {
+                    VariableTree oldVariableTree = (VariableTree) t;
+                    VariableTree newVariableTree = make.Variable(
+                            oldVariableTree.getModifiers(),
+                            oldVariableTree.getName(),
+                            make.Type("var"), // NOI18N
+                            oldVariableTree.getInitializer()
+                    );
+                    newStatements.add(make.asReplacementOf(newVariableTree, oldVariableTree));
+                    
+                    // Check variable tree seperated with ","
+                    TokenSequence<JavaTokenId> tokenSequence = info.getTreeUtilities().tokensFor(t);
+                    tokenSequence.moveEnd();
+                    if (tokenSequence.movePrevious()) {
+                        if (tokenSequence.token().id() != JavaTokenId.COMMA) {
+                            break;
+                        }
+                    }
+                }
+            }
+            if (current + 1 < statements.size()) {
+                newStatements.addAll(statements.subList(current + 1, statements.size()));
+            }
+
+            Tree target;
+
+            switch (parent.getKind()) {
+                case BLOCK:
+                    target = make.Block(newStatements, ((BlockTree) parent).isStatic());
+                    break;
+                //Todo : Handle CASE scenario
+                /*case CASE:
+                    target = make.Case(((CaseTree) parent).getExpression(), newStatements);
+                    break;*/
+                default:
+                    throw new IllegalStateException(parent.getKind().name());
```

Comment:
Since the hint will be shown only for parent.kind BLOCK or CASE, do we need default case. I think we should not throw exception from fix hint.